### PR TITLE
inductor: sort the reads buf by name

### PR DESCRIPTION
**Pull request by chunyuan-w**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #89743
* #90755
* #89742
* __->__ #89744

Sort `read_writes.reads` by name to make sure the same graph is generated for a fixed model. Otherwise, the buffer reuse may be different since the order of `read_writes.reads` is random.

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/_inductor/scheduler.py [hunk 1]
```diff
@@ -318,7 +318,10 @@ def allocate(self):
             from .codegen.triton_template import should_use_template
             from .codegen.wrapper import buffer_reuse_key
 
-            for read in self.read_writes.reads:
+            unordered_reads = list(self.read_writes.reads)
+            ordered_reads = sorted(unordered_reads, key=lambda x: x.name)
```
Comment (jgong5) [inline torch/_inductor/scheduler.py:322]: why not sort on the `self.read_writes.reads` set directly?

Comment (chunyuan-w) [reply torch/_inductor/scheduler.py]: Changed to directly sort on the `self.read_writes.reads`.

**Review by jgong5 — APPROVED** [review]

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**